File: AddressBook/react-client/src/dtos/AddressLabel.ts
import Common from "../utility/Common";
import { PagedReq } from "./PagedReq";

export interface AddressLabelRes {
  addressLabelId?: number;
  label?: string;
}

export class AddressLabelReqEdit {
  label?: string = "";
}

export class AddressLabelReqSearch extends PagedReq {
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
  }
}

File: AddressBook/Dtos/Dtos/AddressLabel.cs
using AddressBook.Common.Paging;
using System.ComponentModel.DataAnnotations;

namespace AddressBook.Dtos
{
    public class AddressLabelRes
    {
        public int AddressLabelId { get; set; }
        public string? Label { get; set; }
    }

    public class AddressLabelReqEdit
    {
        [Required, MaxLength(20)]
        public string? Label { get; set; }
    }

    public class AddressLabelReqSearch : PagedReq
    {

    }
}
